src/ngraph/pass/constant_folding_shape_of.cpp

```diff
@@ -47,6 +49,34 @@ void pass::ConstantFolding::construct_constant_shape_of()
 
             return true;
         }
+        else if (partial_shape.rank().is_static())
+        {
+            auto shape_of = make_shared<op::ShapeOf>(arg_match);
+            auto dimensions = NodeVector{};
+            auto output_dimensions = vector<Dimension>(partial_shape);
+            for (size_t i = 0; i < output_dimensions.size(); ++i)
+            {
+                if (output_dimensions[i].is_static())
+                {
+                    auto value = std::vector<int64_t>{output_dimensions[i].get_length()};
+                    dimensions.push_back(make_shared<op::Constant>(element::i64, Shape{1}, value));
```

Comment:
There is a constructor that takes an initializer list as the 3rd argument. That might be easier than creating a vector. Same comment for below